src/Compilers/CSharp/Portable/Syntax/SyntaxTreeDiagnosticEnumerator.cs

```diff
@@ -11,136 +13,121 @@ namespace Microsoft.CodeAnalysis.CSharp
     /// <summary>
     /// An enumerator for diagnostic lists.
     /// </summary>
-    internal struct SyntaxTreeDiagnosticEnumerator
+    internal static class SyntaxTreeDiagnosticEnumerator
     {
-        private readonly SyntaxTree? _syntaxTree;
-        private NodeIterationStack _stack;
-        private Diagnostic? _current;
-        private int _position;
         private const int DefaultStackCapacity = 8;
 
-        internal SyntaxTreeDiagnosticEnumerator(SyntaxTree syntaxTree, GreenNode? node, int position)
+        public static IEnumerable<Diagnostic> EnumerateDiagnostics(SyntaxTree syntaxTree, GreenNode root, int position)
         {
-            _syntaxTree = null;
-            _current = null;
-            _position = position;
-            if (node != null && node.ContainsDiagnostics)
```

Comment:
no need for null check.  singular caller already checks and throws on null.  THis file is also NRT safe.